static/js/filter.js

```diff
@@ -426,6 +427,147 @@ Filter.prototype = {
+            case 'is-private':
+                return  '/#narrow/is/private';
+            case 'is-starred':
+                return  '/#narrow/is/starred';
+            case 'is-mentioned':
+                return  '/#narrow/is/mentioned';
+            case 'pm-with':
+                // join is used to transform the array to a comma separated string
+                return  '/#narrow/pm-with/' + people.emails_to_slug(this.operands('pm-with').join());
+                // TODO: It is ambiguous how we want to handle the 'sender' case,
+                // we may remove it in the future based on design decisions
+            case 'sender':
+                return  '/#narrow/sender/' + people.emails_to_slug(this.operands('sender')[0]);
+            }
+        }
+
+        return "#"; // redirect to All
+    },
+
+    get_icon: function () {
+        const term_types = this.sorted_term_types();
+        if (term_types.length === 1 && ['in-home', 'in-all'].includes(term_types[0])) {return 'home';}
+        switch (term_types[0]) {
+        case 'stream':
+            if (stream_data.get_sub_by_name(this.operands('stream')[0]).invite_only) {
+                return 'lock';
+            }
+            return 'hashtag';
+        case 'is-private':
+            return 'envelope';
+        case 'is-starred':
+            return 'star';
+        case 'is-mentioned':
+            return 'at';
+        case 'pm-with':
+            return 'envelope';
+        }
+    },
+
+    get_title: function () {
+        const term_types = this.sorted_term_types();
+        if (term_types.length === 1 && ['in-home', 'in-all'].includes(term_types[0])) {return 'home';}
+        if (term_types.length === 3 && _.isEqual(term_types, ['stream', 'topic', 'search'])) {return stream_data.get_sub_by_name(this.operands('stream')[0]).name;}
+        if (term_types.length === 2 && _.isEqual(term_types, ['stream', 'topic'])) {return stream_data.get_sub_by_name(this.operands('stream')[0]).name;}
+        if (term_types.length === 1 || term_types.length === 2 && term_types[1] === 'search') {
+            switch (term_types[0]) {
+            case 'stream':
+                return stream_data.get_sub_by_name(this.operands('stream')[0]).name;
+            case 'is-starred':
+                return  'starred';
+            case 'is-mentioned':
+                return  'mentioned';
+            case 'is-private':
+                return 'Private Messages';
+            }
+            if (term_types[0] === 'pm-with') {
+                const emails = this.operands('pm-with')[0].split(',');
+                const names = emails.map(email => {
+                    if (!people.get_by_email(email)) {
+                        return email;
+                    }
+                    return people.get_by_email(email).full_name;
+                });
+                return  names;
+            }
+        }
+    },
+
+    // get_title_expanded_list: function () {
```

Comment:
Would we want to use this for eg defining the title of the Zulip tab (near the favicon)